docs/custom-stack.asciidoc

```diff
@@ -0,0 +1,262 @@
+So if you prefer, you can set the same configuration options using environment variables:
+
+[source,bash]
+----
+ELASTIC_APM_APP_NAME=<app name>
+ELASTIC_APM_SECRET_TOKEN=<token>
+ELASTIC_APM_SERVER_URL=<server url>
+----
+
+And then just start the agent like so:
+
+[source,js]
+----
+// Start the agent before any thing else in your app
+var apm = require('elastic-apm').start()
+----
+
+See all possible ways to configure the agent <<configuring-the-agent,in the API documentation>>.
+
+[[custom-stack-full-documentation]]
+==== Full documentation
+
+* <<setup,Setup and Configuration>>
+* <<api,API Reference>>
+
+[[custom-stack-performance-monitoring]]
+=== Performance monitoring
+
+Elastic APM automatically measures the performance of your application.
+It records traces for database queries,
+external HTTP requests,
+and other slow operations that happen during requests to your app.
+
+By default the agent will trace <<compatibility,the most common modules>>.
+To trace other events,
+you can use custom traces.
+For information about custom traces,
+see the <<custom-traces,Custom Traces section>>.
+
+Traces are grouped in transactions - by default one for each incoming HTTP request.
+But it's possible to create custom transactions not associated with an HTTP request.
+See the <<custom-transactions,Custom Transactions section>> for details.
+
+[[custom-stack-route-naming]]
+==== Route naming
+
+The Node.js agent tracks incoming HTTP requests to your application in what are called "transactions".
+All transactions with the same name are grouped together automatically.
+
+In a normal web application you want to name transactions based on the route that matches the incoming HTTP request.
+So say you have a route to display posts on a blog identified by `GET /posts/{id}`.
+You want requests `GET /posts/12`, `GET /posts/42` etc to be grouped together under a transaction named `GET /posts/{id}`.
+
+If you are using Express, hapi or koa-router this naming happens automatically based on the names of your routes.
+If you use another framework or a custom router you will see that the transactions are simply grouped together in a few big chunks named "unknown route".
+In that case,
+you will need to help us out a little by supplying a name for each transaction.
+You can do that by calling <<apm-set-transaction-name,`apm.setTransactionName()`>> at any time during the request with the name of the transaction as the first argument.
+
+Example application using the https://github.com/watson/patterns[patterns] module for route handling:
+
+[source,js]
+----
+var apm = require('elastic-apm').start()
+var http = require('http')
+var patterns = require('patterns')()
+
+// Setup routes and their respective route handlers
+patterns.add('GET /', require('./routes/index'))
+patterns.add('GET /posts', require('./routes/posts').index)
+patterns.add('GET /posts/{id}', require('./routes/posts').show)
```

Comment:
`routes/index` and `routes/post` does not exist. The example looks to me like I could simply copy & paste it and it should work. It doesn't unfortunately. 
This might not be an issue in the real world where users have their own apps they instrument. However I still recommend to rewrite the example or make it clear it is just a snippet.

Reply:
Fixed by changing the paragraph just before the code example